### PR TITLE
Docs: Add RBAC role definition, include and route action protection

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:

This PR adds the plugin role registrations and RBAC action protection fields to the documentation.

Since we want to enable the `accessControlOnCall` feature, I think it's time to make this official.

**Which issue(s) this PR fixes**:

Follow up to https://github.com/grafana/grafana/pull/80592

**Special notes for your reviewer**:

### CODE REVIEW

**Review by josmperez — DISMISSED** [review]
LGTM.

File: docusaurus/docs/metadata.md [hunk 1]
```diff
@@ -207,6 +209,29 @@ For data source plugins. There is a query options section in the plugin's query
...
+List of RBAC roles defined by the plugin and their default assignments to basic roles (`Viewer`, `Editor`, `Admin`, `Grafana Admin`).
+
+Requires Grafana version 9.4.0 or later. Currently, this is behind the `accessControlOnCall` feature toggle.
+
+### Properties
+
+| Property | Type            | Required | Description                                                                                           |
+| -------- | --------------- | -------- | ----------------------------------------------------------------------------------------------------- |
+| `role`   | [object](#role) | **yes**  | RBAC role definition to bundle related RBAC permissions on the plugin.                                |
+| `grant`  | string[]        | no       | Default assignments of the role to Grafana basic roles (`Viewer`, `Editor`, `Admin`, `Grafana Admin`) |
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:223]:
```suggestion
| `grant`  | string[]        | No       | Default assignments of the role to Grafana basic roles (`Viewer`, `Editor`, `Admin`, `Grafana Admin`) |
```

File: docusaurus/docs/metadata.md [hunk 2]
```diff
@@ -207,6 +209,29 @@ For data source plugins. There is a query options section in the plugin's query
...
+
+### role
+
+A role bundle a set of related RBAC permissions on your plugin. The RBAC actions defined in your role must start with your plugin id (ex: `grafana-test-app.projects:read`).
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:227]: "A role bundle a set of" is acwward, can you rephrase this?

Comment (josmperez) [inline docusaurus/docs/metadata.md:227]:
```suggestion
A role bundle a set of related RBAC permissions on your plugin. The RBAC actions defined in your role must start with your plugin `id` (ex: `grafana-test-app.projects:read`).
```

File: docusaurus/docs/metadata.md [hunk 3]
```diff
@@ -207,6 +209,29 @@ For data source plugins. There is a query options section in the plugin's query
...
+
+| Property      | Type                     | Required | Description                    |
+| ------------- | ------------------------ | -------- | ------------------------------ |
+| `name`        | string                   | **yes**  | Display name of the role.      |
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:231]:
```suggestion
| `name`        | string                   | **Yes**  | Displays name of the role.      |
```

File: docusaurus/docs/metadata.md [hunk 4]
```diff
@@ -207,6 +209,29 @@ For data source plugins. There is a query options section in the plugin's query
...
+| `description` | string                   | **yes**  | Describe the aim of the role.  |
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:232]:
```suggestion
| `description` | string                   | **Yes**  | Describes the aim of the role.  |
```

File: docusaurus/docs/metadata.md [hunk 5]
```diff
@@ -207,6 +209,29 @@ For data source plugins. There is a query options section in the plugin's query
...
+| `permissions` | [object](#permissions)[] | **yes**  | RBAC permission on the plugin. |
```
Comment (josmperez) [inline docusaurus/docs/metadata.md:233]:
```suggestion
| `permissions` | [object](#permissions)[] | **Yes**  | RBAC permission on the plugin. |
```

**Review by josmperez — CHANGES_REQUESTED** [review]
Some minor style issues.

**Review by josmperez — APPROVED** [review]
LGTM.

File: docusaurus/docs/metadata.md [hunk 6]
```diff
@@ -207,6 +209,30 @@ For data source plugins. There is a query options section in the plugin's query
...
+
+A role groups your plugin's related RBAC permissions (ex: `Projects Admin` would group permissions to create, read, write and delete projects).
+The RBAC actions defined in your role must start with your plugin `id` (ex: `grafana-test-app.projects:read`).
+
+| Property      | Type                     | Required | Description                    |
+| ------------- | ------------------------ | -------- | ------------------------------ |
+| `name`        | string                   | **Yes**  | Name of the role.              |
+| `description` | string                   | No       | Describes the aim of the role. |
+| `permissions` | [object](#permissions)[] | No       | RBAC permission on the plugin. |
+
```
Comment (leventebalogh) [inline docusaurus/docs/metadata.md:235]: **Nit:** maybe it would make sense to include a small example below for a hypothetical use-case?

**Review by leventebalogh — APPROVED** [review]

File: docusaurus/docs/metadata.md [hunk 7]
```diff
@@ -111,6 +112,7 @@ To try this feature out, follow this [example](https://github.com/grafana/grafan
 
 | Property     | Type    | Required | Description                                                                                                                                                                                     |
 | ------------ | ------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `action`     | string  | No       | The RBAC action a user must have to see this page in the navigation menu.                                                                                                                       |
```
Comment (academo) [inline docusaurus/docs/metadata.md:115]:
```suggestion
| `action`     | string  | No       | The RBAC action an user must have to see this page in the navigation menu.                                                                                                                       |
```

File: docusaurus/docs/metadata.md (hunk 6 above)
Comment (gamab) [reply docusaurus/docs/metadata.md:235]: I think it makes sense.
I'm trying to write an example plugin: https://github.com/grafana/grafana-plugin-examples/pull/242
I'll add it to the roles section.
```md
To try this feature out, follow this [example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-rbac/README.md)
```
Wdyt, should I put this PR on hold and wait for the example plugin to be merged?
Or should I merge this one, we track that comment in an issue, then add the link?
